### PR TITLE
Implement magnetometer-based on-canvas compass

### PR DESCRIPTION
**Pull request by nirvn**
This PR implements a proper magnetometer-based on-canvas compass in QField. The feature requires devices to have a magnetometer sensor hardware (PSA: when you buy a mobile device, make sure you check if that hardware is present or not, many entry level devices leave that piece of hardware out to lower production cost).

The sensor is only turned on whenever the positioning is active to insure we're not draining battery needlessly.

The visual compass looks like this:
image

(Temporarily include #2539 until that is merged)

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f7614c7a5b617685793c73f272b5f99e069438c6:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-e8d74ebac245e6032ba195e172f34c48bf655f76-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-e8d74ebac245e6032ba195e172f34c48bf655f76-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-e8d74ebac245e6032ba195e172f34c48bf655f76-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-e8d74ebac245e6032ba195e172f34c48bf655f76-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0da17f202212dd56f48511ffcec3e1eacdc396e5:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-8f78513e7bde4289d7d659188315d7fa8589ba30-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-8f78513e7bde4289d7d659188315d7fa8589ba30-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-8f78513e7bde4289d7d659188315d7fa8589ba30-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-8f78513e7bde4289d7d659188315d7fa8589ba30-x86-android.apk)

**Review by suricactus — APPROVED** [review]

File: src/qml/LocationMarker.qml [hunk 1]
```diff
@@ -31,21 +49,51 @@ Item {
...
+      startY: 0
+      PathLine { x: 18; y: 20 }
+      PathLine { x: 10; y: 14 }
+      PathLine { x: 2; y: 20 }
+      PathLine { x: 10; y: 0 }
+
+      SequentialAnimation on fillColor  {
+        loops: Animation.Infinite
+        ColorAnimation  { from: Theme.positionColor; to: Theme.darkPositionColor; duration: 2000; easing.type: Easing.InOutQuad }
+        ColorAnimation  { from: "#2374b5"; to: Theme.positionColor; duration: 1000; easing.type: Easing.InOutQuad }
```
[the remark this answers is not in the document]
Comment (nirvn) [reply src/qml/LocationMarker.qml:94]: @suricactus , oups, na, that's just an omitting on my part.

**Review by suricactus — APPROVED** [review]
Thanks! Works well.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 4073667e87d661b784b2e453bf4a7bef102ffc15:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-f91166ed135ee73403add24a1be8894c1b972a21-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-f91166ed135ee73403add24a1be8894c1b972a21-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-f91166ed135ee73403add24a1be8894c1b972a21-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2541-f91166ed135ee73403add24a1be8894c1b972a21-x86-android.apk)

**Pull request merged**